internal/system/agent/services.go

```diff
@@ -34,18 +37,110 @@ const (
 	RESTART = "restart"
 )
 
+func InvokeMetrics(services []string, ctx context.Context) (MetricsRespMap, error) {
+
+	m := MetricsRespMap{}
+	m.Metrics = map[string]interface{}{}
+
+	// Loop through requested actions, along with (any) respectively-supplied parameters.
+	for _, service := range services {
+		LoggingClient.Debug("invoke metrics")
+
+		if Configuration.MetricsMechanism == "direct-service" {
+			es := ExecuteService{}
+			out, err := es.Metrics(ctx, service)
+			if err != nil {
+				LoggingClient.Error("error fetching metrics")
+				return m, err
+			} else {
+				m.Metrics[service] = ProcessResponse(string(out))
+			}
+		} else if Configuration.MetricsMechanism == "executor" {
+			ea := ExecuteApp{}
+			out, err := ea.Metrics(ctx, service)
+			if err != nil {
+				LoggingClient.Error("error fetching metrics")
+				return m, err
+			} else {
+				result, _ := processOutput(out)
+				m.Metrics[service] = ProcessResponse(result)
+			}
+		} else if Configuration.MetricsMechanism == "custom" {
+			err := fmt.Errorf("the requested custom executor (e.g. snap) has not been integrated")
+			LoggingClient.Error(err.Error())
+			m.Metrics[service] = fmt.Sprintf(err.Error())
+		} else {
+			err := fmt.Errorf("the requested metrics mechanism is not supported")
+			LoggingClient.Error(err.Error())
+		}
+	}
+	return m, nil
+}
+
+func processOutput(bytes []byte) (string, error) {
+
+	s := string(bytes)
+	lines, err := stringToLines(s)
+
+	if err != nil {
+		LoggingClient.Error(err.Error())
+		return "", err
+	}
+	relevantLineReturned, err := findRelevantLines(lines)
+	if err != nil {
+		return "", fmt.Errorf(err.Error())
+	}
+	return relevantLineReturned, nil
+}
+
+func findRelevantLines(allLines []string) (string, error) {
+
+	var all []string
+	var proceed bool
+	proceed = false
+
+	for _, line := range allLines {
+
+		if strings.Contains(line, "success performing metrics on") {
+			proceed = true
+		}
+		if proceed {
+			all = append(all, line)
+		}
```

Comment:
`proceed` is superfluous. It's also not reset to `false` between iterations. This check can be collapsed like so
```
if strings.Contains(line, "success performing metrics on") {
    all = append(all, line)
}
```